fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java

```diff
@@ -183,6 +185,44 @@ public DatatableData retrieveDatatable(final String datatable) {
         return datatableData;
     }
 
+    @Override
+    public List<JsonObject> queryDataTable(String datatable, String columnFilter, String valueFilter, String resultColumns) {
+        Arrays.asList(datatable, columnFilter, valueFilter, resultColumns).forEach(SQLInjectionValidator::validateDynamicQuery);
+        if(columnFilter == null || valueFilter == null) {
```

Comment:
I think it would be better to have a validation error if the filter column or filter value is not provided.

Reply:
changing the code accordingly